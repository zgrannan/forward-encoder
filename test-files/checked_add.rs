#[analyzer::run]
#[allow(dead_code)]
fn foo(x: u32) -> u32 {
    if x + 5 < 3{
        1
    } else {
        2
    }
}

fn main(){}
