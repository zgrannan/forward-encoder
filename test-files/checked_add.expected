Analyzing file test-files/checked_add.rs
Function foo():
Terminal block bb4: let _4 = _1 in
let _5 = CheckedAdd(_4, const 5_u32) in
let _3 = move (_5.0: u32) in
let _2 = Lt(move _3, const 3_u32) in
let _0 = if move _2 != Int(0) then const 1_u32 else const 2_u32 in
_0
